contracts/ERC721A.sol

```diff
@@ -18,25 +18,37 @@ import '@openzeppelin/contracts/utils/introspection/ERC165.sol';
  *
  * Assumes serials are sequentially minted starting at 0 (e.g. 0, 1, 2, 3..).
  *
- * Does not support burning tokens to address(0).
+ * Assumes that an owner cannot have more than 2**64 - 1 (max value of uint64) of supply.
  *
- * Assumes that an owner cannot have more than the 2**128 - 1 (max value of uint128) of supply
+ * Assumes that the maximum token id cannot exceed 2**128 - 1 (max value of uint128).
  */
 contract ERC721A is Context, ERC165, IERC721, IERC721Metadata, IERC721Enumerable {
     using Address for address;
     using Strings for uint256;
 
+    // Compiler will pack this into a single 256bit word.
     struct TokenOwnership {
+        // The address of the owner.
         address addr;
+        // Keeps track of the start time of ownership with minimal overhead for tokenomics.
         uint64 startTimestamp;
+        // Whether the token has been burned.
+        bool burned;
     }
 
+    // Compiler will pack this into a single 256bit word.
     struct AddressData {
-        uint128 balance;
-        uint128 numberMinted;
+        // Realistically, 2**64-1 is more than enough.
+        uint64 balance;
+        // Keeps track of mint count with minimal overhead for tokenomics.
+        uint64 numberMinted;
+        // Keeps track of burn count with minimal overhead for tokenomics.
+        uint64 numberBurned;
```

Comment:
Any reason why you decided to include this?

Reply:
Keeping track of numberMinted and the block timestamps are not in the spec, but is useful to make use of the remaining bytes. Same for numberBurned.

If you don’t like it, it can be removed.

Reply:
We can keep this in for now - was just curious. We're still going through this PR, just been a super busy week. Will try to get this done soon!

Reply:
Of the currently opened feature PRs, which ones likely to be merged before this PR?

Reply:
Merged the custom errors from https://github.com/chiru-labs/ERC721A/pull/73.

Reply:
This is the next PR I'm reviewing